docs/fastdds/dds_layer/core/status/status.rst

```diff
@@ -3,3 +3,630 @@
+| total_count              | ``int32_t``                             |
++--------------------------+-----------------------------------------+
+| total_count_change       | ``int32_t``                             |
++--------------------------+-----------------------------------------+
+| last_reason              | ``SampleRejectedStatusKind``            |
++--------------------------+-----------------------------------------+
+| last_instance_handle     | ``InstanceHandle_t``                    |
++--------------------------+-----------------------------------------+
+
+* **total_count**: Total cumulative count of rejected samples under the :ref:`dds_layer_topic_topic`
+  of the current :ref:`dds_layer_subscriber_dataReader`.
+
+* **total_count_change**: The change in **total_count** since
+  the last time |on_sample_rejected| was called or the status was read.
+  It can only be positive or zero.
+
+* **last_reason**: The reason for rejecting the last rejected sample.
+  If no sample was ever rejected, it will have value ``NOT_REJECTED``.
+  See :ref:`dds_layer_core_status_sampleRejectedStatusKind` for further details.
+
+* **last_instance_handle**: Handle to the last instance whose sample was rejected.
+  If no sample was ever rejected, it will have value ``c_InstanceHandle_Unknown``.
+
+.. warning::
+
+    Currently this status is not supported and will be implemented in future releases.
+    As a result, trying to access this status will return ``NOT_SUPPORTED``
+    and the corresponding listener will never be called.
+
+
+.. _dds_layer_core_status_sampleRejectedStatusKind:
+
+SampleRejectedStatusKind
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are four possible values:
+
+* ``NOT_REJECTED``: It means there have been no rejections so far on this :ref:`dds_layer_subscriber_dataReader`.
+  The rejection reason will have this value only while the total count of rejections is zero.
+* ``REJECTED_BY_INSTANCES_LIMIT``: The sample was rejected because the
+  :ref:`max_instances<resourcelimitsqospolicy>` limit was reached.
+* ``REJECTED_BY_SAMPLES_LIMIT``: The sample was rejected because the
+  :ref:`max_samples<resourcelimitsqospolicy>` limit was reached.
+* ``REJECTED_BY_SAMPLES_PER_INSTANCE_LIMIT``: The sample was rejected because the
+  :ref:`max_samples_per_instance<resourcelimitsqospolicy>` limit was reached.
+
+
+.. _dds_layer_core_status_subscriptionMatchedStatus:
+
+SubscriptionMatchedStatus
+-------------------------
+
+This status changes every time the :ref:`dds_layer_subscriber_dataReader` finds a :ref:`dds_layer_publisher_dataWriter`
+that matches the :ref:`dds_layer_topic_topic` and has a common partition and a compatible QoS,
+or has ceased to be matched with a :ref:`dds_layer_publisher_dataWriter` that was previously considered to be matched.
+It is also changed when a matched :ref:`dds_layer_publisher_dataWriter` has changed its
+:ref:`dds_layer_publisher_dataWriterQos`.
```

Comment:
Before, there was a way to know whether the callback was triggered by a new data writer or by a change in one of the matched ones. How is that done now? Just by looking at the counters?

Reply:
specifically looking at the count_change. Is it +1 (new match) or -1 (new unmatch)

Reply:
I removed the sentence about the status changing when the remote QoS changes because I don't see that happening in the code, if the QoS changes but there is still a match, nothing seems to be called.